git/objects/submodule/base.py

```diff
@@ -1184,8 +1184,9 @@ def iter_items(cls, repo, parent_commit='HEAD'):
                     entry = index.entries[index.entry_key(p, 0)]
                     sm = Submodule(repo, entry.binsha, entry.mode, entry.path)
                 except KeyError:
-                    raise InvalidGitRepositoryError(
```

Comment:
Is there a way for this Exception to stay, just as there is a test that expects this behaviour (see the test-failure on travis).

Reply:
If we do that, we wouldn't be able to keep listing the submodules here.

Reply:
@stsewd I think all that's left before the merge is to fix the now failing test. It could be changed to now expect to no failure and to see a listing of the existing submodules - if you think there is not too much overlap with the test you added.